app/controllers/api/v0/versions_controller.rb

```diff
@@ -25,6 +27,41 @@ def show
     }
   end
 
+  def raw
+
+    @version ||= version_collection.find(params[:id])
+
+    def self.public_hosts=(hosts)
+      hosts = [] if hosts.nil?
+      hosts = hosts.split(' ') if hosts.is_a?(String)
+      unless hosts.is_a?(Enumerable) && hosts.all? {|host| host.is_a?(String)}
+        raise StandardError, 'Public hosts must be a string or enumerable of strings'
+      end
+      @public_hosts = hosts
+    end
+
+    if ENV['PUBLIC_ARCHIVE_HOSTS']
+      self.public_hosts = ENV['PUBLIC_ARCHIVE_HOSTS']
+    end
+
+    def self.public_hosts
+      @public_hosts || []
+    end
+
+    def self.public_archive_uri?(uri)
+      public_hosts.any? {|base| uri.starts_with?(base)}
+    end
```

Comment:
Are all these inline methods meant to ultimately be here or is this just proof-of-concept? This definitely isn’t mergeable like this.

These methods should at least be part of the class definition, if not separated into a concern/mixin or even a separate service class or module (like `Archiver`, perhaps—or maybe this logic belongs *in* archiver).

Reply:
Yeah, this is definitely a sketch that needs to be cleaned up.

Reply:
I moved pretty much everything into Archiver.